core/src/operations/decorators/mod.rs

```diff
@@ -0,0 +1,48 @@
+mod advice;
+use crate::utils::collections::Vec;
+pub use advice::AdviceInjector;
+use core::fmt;
+
+// DECORATORS
+// ================================================================================================
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Decorator {
+    /// Injects zero or more values at the head of the advice tape as specified by the injector.
+    /// This operation affects only the advice tape, but has no effect on other VM components
+    /// (e.g., stack, memory), and does not advance VM clock.
+    Advice(AdviceInjector),
+}
+
+impl fmt::Display for Decorator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Advice(injector) => write!(f, "advice({})", injector),
+        }
+    }
+}
+
+pub type DecoratorMap = Vec<(usize, Decorator)>;
```

Comment:
I'm not sure we need this type any more, and technically, it is not really a map - so, if we do want to have a type alias, probably a better name is needed.

Reply:
I think it might be better to keep the type alias since it is being used in many places and with the addition of other decorators in the future, it makes sense to have a specified type alias for this? However, I can't think of a better name.

Reply:
Got it! Let's name it `DecoratorList` then. Still not ideal, but I think better than `DecoratorMap`.

Reply:
Done